File: backend/ai/factory.py
import os
from ai.base import AIProvider
from ai.openai_provider import OpenAIProvider
from ai.gemini_provider import GeminiProvider


def get_ai_provider() -> AIProvider:
    """Провайдер задаётся переменной AI_PROVIDER=openai|gemini. По умолчанию openai."""
    provider = (os.getenv("AI_PROVIDER") or "openai").strip().lower()
    if provider == "gemini":
        return GeminiProvider()
    return OpenAIProvider()

File: backend/ai/factory.js
const OpenAIProvider = require("./openai-provider");
const GeminiProvider = require("./gemini-provider");

function getAIProvider() {
  const provider = (process.env.AI_PROVIDER || "openai").trim().toLowerCase();
  if (provider === "gemini") return new GeminiProvider();
  return new OpenAIProvider();
}

module.exports = { getAIProvider };
